python/cuml/metrics/confusion_matrix.py

```diff
@@ -0,0 +1,113 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import numpy as np
+import cupy as cp
+
+from cuml.utils import input_to_cuml_array
+from cuml.utils.memory_utils import with_cupy_rmm
+from cuml.metrics.utils import sorted_unique_labels
+from cuml.prims.label import make_monotonic
+
+
+@with_cupy_rmm
+def confusion_matrix(y_true, y_pred,
+                     labels=None,
+                     sample_weight=None,
+                     normalize=None):
+    """Compute confusion matrix to evaluate the accuracy of a classification.
+
+    Parameters
+    ----------
+    y_true : array-like (device or host) shape = (n_samples,)
+        or (n_samples, n_outputs)
+        Ground truth (correct) target values.
+    y_pred : array-like (device or host) shape = (n_samples,)
+        or (n_samples, n_outputs)
+        Estimated target values.
+    labels : array-like (device or host) shape = (n_classes,), optional
+        List of labels to index the matrix. This may be used to reorder or
+        select a subset of labels. If None is given, those that appear at least
+        once in y_true or y_pred are used in sorted order.
+    sample_weight : array-like (device or host) shape = (n_samples,), optional
+        Sample weights.
+    normalize : string in [‘true’, ‘pred’, ‘all’]
+        Normalizes confusion matrix over the true (rows), predicted (columns)
+        conditions or all the population. If None, confusion matrix will not be
+        normalized.
+
+    Returns
+    -------
+    C : array-like (device or host) shape = (n_classes, n_classes)
+        Confusion matrix.
+    """
+    y_true, n_rows, n_cols, dtype = \
+        input_to_cuml_array(y_true, check_dtype=[cp.int32, cp.int64])
+
+    y_pred, _, _, _ = \
+        input_to_cuml_array(y_pred, check_dtype=dtype,
+                            check_rows=n_rows, check_cols=n_cols)
+
+    if labels is None:
+        labels = sorted_unique_labels(y_true, y_pred)
+        n_labels = len(labels)
+    else:
+        labels, n_labels, _, _ = \
+            input_to_cuml_array(labels, check_dtype=dtype, check_cols=1)
+        labels = labels.to_output('cupy')
+    if sample_weight is None:
+        sample_weight = cp.ones(n_rows, dtype=dtype)
```

Comment:
instead of cp.ones i think it will be good to use cumlarray.ones

Reply:
As explained in the following comment `sample_weight` has to be a `cupy.ndarray` because of the call to `cp.sparse.coo_matrix`:
https://github.com/rapidsai/cuml/blob/81ed57a80701064c0e227d257401b29715793af0/python/cuml/metrics/confusion_matrix.py#L100